### ISSUE TITLE
Issues zooming in and out with shared axis

### ISSUE THREAD

**Issue by cimarronm** [matplotlib/matplotlib#5270]
I'm noticing a problem when zooming in on plots with shared axis and number of points >1000. If I zoom in on a plot and then try going back ('r' or 'h' key), the plots seem to have much of the data masked out.

##### Minimum working example
```python
import matplotlib.pyplot as plt
import numpy as np

fig, ax = plt.subplots(nrows=2, ncols=1, sharex=True)

size = 10000
ax[0].plot(np.random.random(size))
ax[1].plot(np.arange(size))

plt.show()
```

**Comment by QuLogic** [comment]
I don't think the shared axis is even necessary; simply this will trigger it:
```python
import matplotlib.pyplot as plt
import numpy as np
plt.plot(np.random.random(10000))
plt.show()
```

**Comment by mdboom** [comment]
Wow.  That's pretty serious.  I'm going to look into it.

**Comment by mdboom** [comment]
First bad commit is: 3c2c0370964e5ee581fc63343f67c22f140e7fbc

**Comment by mdboom** [comment]
Would you mind trying #5275 for a fix?

**Comment by cimarronm** [comment]
@mdboom, #5275 does fix it for me. :+1:  Thanks for looking into it; I didn't have time yesterday but wanted to make sure I put the issue up.

**Comment by mdboom** [comment]
No problem -- it's always better to know about the problem, and thankfully the fix for this should make it into 1.5.0!

**Comment by QuLogic** [comment]
#5275 was merged.
